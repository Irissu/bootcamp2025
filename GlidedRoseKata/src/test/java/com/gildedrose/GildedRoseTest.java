package com.gildedrose;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class GildedRoseTest {

    @Test
    void foo() {
        Item[] items = new Item[] { new Item("foo", 0, 0) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        assertEquals("foo", app.items[0].name);
    }

    @Test
    void testProductNameIsConjured() {
        Item[] items = new Item[] { new Item("Conjured Space Cake", 0, 0) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        assertTrue(items[0].name.contains("Conjured"));
    }

    @Test
    void testQualityDegradationOfConjuredItems() {
        Item[] items = new Item[] {
            new Item("Conjured Space Brownie", 3, 5),
            new Item("Conjured Space Cake", 1, 10),
            new Item("Conjured Split Banana", 0, 6)
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        assertEquals(3, items[0].quality);
        assertEquals(8, items[1].quality);
        assertEquals(2, items[2].quality);
    }

    @Test
    void testCheckPositiveQualityOfConjuredItems(){
        Item[] items = new Item[] { new Item("Conjured Space Cake", 0, 0) };

        GildedRose app = new GildedRose(items);
        app.updateQuality();

        assertTrue(items[0].quality >= 0);
    }
}
