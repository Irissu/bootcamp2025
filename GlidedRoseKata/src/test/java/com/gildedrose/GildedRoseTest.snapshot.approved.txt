day,name, sellIn, quality
0,+5 Dexterity Vest, 10, 20
0,Aged Brie, 2, 0
0,Elixir of the Mongoose, 5, 7
0,Sulfuras, Hand of Ragnaros, 0, 80
0,Sulfuras, Hand of Ragnaros, -1, 80
0,Backstage passes to a TAFKAL80ETC concert, 15, 20
0,Backstage passes to a TAFKAL80ETC concert, 10, 49
0,Backstage passes to a TAFKAL80ETC concert, 5, 49
0,Conjured Mana Cake, 3, 6
1,+5 Dexterity Vest, 9, 19
1,Aged Brie, 1, 1
1,Elixir of the Mongoose, 4, 6
1,Sulfuras, Hand of Ragnaros, 0, 80
1,Sulfuras, Hand of Ragnaros, -1, 80
1,Backstage passes to a TAFKAL80ETC concert, 14, 21
1,Backstage passes to a TAFKAL80ETC concert, 9, 50
1,Backstage passes to a TAFKAL80ETC concert, 4, 50
1,Conjured Mana Cake, 2, 4
2,+5 Dexterity Vest, 8, 18
2,Aged Brie, 0, 2
2,Elixir of the Mongoose, 3, 5
2,Sulfuras, Hand of Ragnaros, 0, 80
2,Sulfuras, Hand of Ragnaros, -1, 80
2,Backstage passes to a TAFKAL80ETC concert, 13, 22
2,Backstage passes to a TAFKAL80ETC concert, 8, 50
2,Backstage passes to a TAFKAL80ETC concert, 3, 50
2,Conjured Mana Cake, 1, 2
3,+5 Dexterity Vest, 7, 17
3,Aged Brie, -1, 4
3,Elixir of the Mongoose, 2, 4
3,Sulfuras, Hand of Ragnaros, 0, 80
3,Sulfuras, Hand of Ragnaros, -1, 80
3,Backstage passes to a TAFKAL80ETC concert, 12, 23
3,Backstage passes to a TAFKAL80ETC concert, 7, 50
3,Backstage passes to a TAFKAL80ETC concert, 2, 50
3,Conjured Mana Cake, 0, 0
4,+5 Dexterity Vest, 6, 16
4,Aged Brie, -2, 6
4,Elixir of the Mongoose, 1, 3
4,Sulfuras, Hand of Ragnaros, 0, 80
4,Sulfuras, Hand of Ragnaros, -1, 80
4,Backstage passes to a TAFKAL80ETC concert, 11, 24
4,Backstage passes to a TAFKAL80ETC concert, 6, 50
4,Backstage passes to a TAFKAL80ETC concert, 1, 50
4,Conjured Mana Cake, -1, 0
5,+5 Dexterity Vest, 5, 15
5,Aged Brie, -3, 8
5,Elixir of the Mongoose, 0, 2
5,Sulfuras, Hand of Ragnaros, 0, 80
5,Sulfuras, Hand of Ragnaros, -1, 80
5,Backstage passes to a TAFKAL80ETC concert, 10, 25
5,Backstage passes to a TAFKAL80ETC concert, 5, 50
5,Backstage passes to a TAFKAL80ETC concert, 0, 50
5,Conjured Mana Cake, -2, 0
6,+5 Dexterity Vest, 4, 14
6,Aged Brie, -4, 10
6,Elixir of the Mongoose, -1, 0
6,Sulfuras, Hand of Ragnaros, 0, 80
6,Sulfuras, Hand of Ragnaros, -1, 80
6,Backstage passes to a TAFKAL80ETC concert, 9, 27
6,Backstage passes to a TAFKAL80ETC concert, 4, 50
6,Backstage passes to a TAFKAL80ETC concert, -1, 0
6,Conjured Mana Cake, -3, 0
7,+5 Dexterity Vest, 3, 13
7,Aged Brie, -5, 12
7,Elixir of the Mongoose, -2, 0
7,Sulfuras, Hand of Ragnaros, 0, 80
7,Sulfuras, Hand of Ragnaros, -1, 80
7,Backstage passes to a TAFKAL80ETC concert, 8, 29
7,Backstage passes to a TAFKAL80ETC concert, 3, 50
7,Backstage passes to a TAFKAL80ETC concert, -2, 0
7,Conjured Mana Cake, -4, 0
8,+5 Dexterity Vest, 2, 12
8,Aged Brie, -6, 14
8,Elixir of the Mongoose, -3, 0
8,Sulfuras, Hand of Ragnaros, 0, 80
8,Sulfuras, Hand of Ragnaros, -1, 80
8,Backstage passes to a TAFKAL80ETC concert, 7, 31
8,Backstage passes to a TAFKAL80ETC concert, 2, 50
8,Backstage passes to a TAFKAL80ETC concert, -3, 0
8,Conjured Mana Cake, -5, 0
9,+5 Dexterity Vest, 1, 11
9,Aged Brie, -7, 16
9,Elixir of the Mongoose, -4, 0
9,Sulfuras, Hand of Ragnaros, 0, 80
9,Sulfuras, Hand of Ragnaros, -1, 80
9,Backstage passes to a TAFKAL80ETC concert, 6, 33
9,Backstage passes to a TAFKAL80ETC concert, 1, 50
9,Backstage passes to a TAFKAL80ETC concert, -4, 0
9,Conjured Mana Cake, -6, 0
